.github/workflows/fpm-deployment.yml

```
@@ -0,0 +1,50 @@
+name: fpm-deployment
+
+on: [push, pull_request]
+env:
+  GCC_V: "10"
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout 🛎️
+      uses: actions/checkout@v2.3.1
+
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.x
+
+    - name: Install fypp
+      run: pip install --upgrade fypp
+
+    - name: Generate stdlib-fpm package 🔧
+      run: |
+        sh ./ci/fpm-deployment.sh
+
+    - name: Install GFortran
+      run: |
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-${{ env.GCC_V }} 100 \
+        --slave /usr/bin/gfortran gfortran /usr/bin/gfortran-${{ env.GCC_V }} \
+        --slave /usr/bingcov gcov /usr/bin/gcov-${{ env.GCC_V }}
+
+    - name: Install fpm latest release
+      uses: fortran-lang/setup-fpm@v3
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Run fpm test ⚙
+      run: |
+        cp -r stdlib-fpm stdlib-fpm-test
+        cd stdlib-fpm-test
+        fpm test
+        fpm test --profile release
+
+    # Update and deploy the f90 files generated by github-ci to the `stdlib-fpm` branch.
+    - name: Deploy 🚀
+      uses: JamesIves/github-pages-deploy-action@4.1.5
```

In the [`stdlib-fpm`](https://github.com/LKedward/stdlib-fpm) workflow, I use the following conditional to prevent deployment for PRs. (This should fix the CI error for this PR)

```suggestion
      uses: JamesIves/github-pages-deploy-action@4.1.5
      if: github.event_name != 'pull_request'
```